src/tracer/MARBL_tracers.F90

```diff
@@ -0,0 +1,734 @@
+
+      ! These are okay, but need option to come in from coupler
+      if (CS%xco2_ind > 0) marbl_instances%surface_flux_forcings(CS%xco2_ind)%field_0d(1) = CS%atm_co2_const
+      if (CS%xco2_alt_ind > 0) marbl_instances%surface_flux_forcings(CS%xco2_alt_ind)%field_0d(1) = CS%atm_alt_co2_const
+
+      !        These fields are receiving physically-reasonable values
+      if (CS%dust_dep_ind > 0) marbl_instances%surface_flux_forcings(CS%dust_dep_ind)%field_0d(1) = 0 ! pop gets from coupler
+      if (CS%fe_dep_ind > 0) marbl_instances%surface_flux_forcings(CS%fe_dep_ind)%field_0d(1) = 0 ! pop gets from coupler (but can read from file)
+
+      !        Read these from /glade/work/mlevy/cesm_inputdata/ndep_ocn_1850_w_nhx_emis_MOM_tx0.66v1_c200827.nc
+      ! MOM_read_data()
+      if (CS%nox_flux_ind > 0) marbl_instances%surface_flux_forcings(CS%nox_flux_ind)%field_0d(1) = 0 ! pop gets from file
+      if (CS%nhy_flux_ind > 0) marbl_instances%surface_flux_forcings(CS%nhy_flux_ind)%field_0d(1) = 0 ! pop gets from file
+
+      !     ii. tracers at surface
+      do m=1,CS%ntr
+        marbl_instances%tracers_at_surface(1,m) = CS%tr(i,j,1,m)
+      end do
+
+      !     iii. surface flux saved state
+      do m=1,size(marbl_instances%surface_flux_saved_state%state)
+        marbl_instances%surface_flux_saved_state%state(m)%field_2d(1) = CS%surface_flux_saved_state(m)%field_2d(i,j)
+      end do
+
+      ! (2) Compute surface fluxes in MARBL
+      call marbl_instances%surface_flux_compute()
+      if (marbl_instances%StatusLog%labort_marbl) then
+        call marbl_instances%StatusLog%log_error_trace("marbl_instances%surface_flux_compute()", &
+                                                       "MARBL_tracers_column_physics")
+        call print_marbl_log(MARBL_instances%StatusLog)
+      end if
+
+      ! (3) Copy output that MOM6 needs to hold on to
+      !     i. saved state
+      !     ii. diagnostics
+      do m=1,size(marbl_instances%surface_flux_diags%diags)
+        ! All diags are 2D coming from surface
+        CS%surface_flux_diags(m)%field_2d(i,j) = real(marbl_instances%surface_flux_diags%diags(m)%field_2d(1))
+      end do
+      do m=1,size(marbl_instances%surface_flux_saved_state%state)
+        CS%surface_flux_saved_state(m)%field_2d(i,j) = marbl_instances%surface_flux_saved_state%state(m)%field_2d(1)
+      end do
+    end do
+  end do
+  ! All diags are 2D coming from surface
+  do m=1,size(CS%surface_flux_diags)
+!    if (CS%surface_flux_diags(m)%id > 0) then
+!      write(log_message, *) "MNL MNL: max diag value ", maxval(abs(CS%surface_flux_diags(m)%field_2d)), &
+!                            " (m = ", m, ")"
+!      call MOM_error(WARNING, log_message)
+!    end if
+    if (CS%surface_flux_diags(m)%id > 0) &
+      call post_data(CS%surface_flux_diags(m)%id, CS%surface_flux_diags(m)%field_2d, CS%diag)
+  end do
+
+  if (present(evap_CFL_limit) .and. present(minimum_forcing_depth)) then
```

Comment:
I wonder if you shouldn't do the vertical diffusion and concentration/dilution by freshwater fluxes first. Also note that the layer thicknesses passed into MARB (if it needs them) should be `h_work` AFTER this routine has been called.